pkg/bundle/publish/catalog_images.go

```diff
@@ -148,7 +148,10 @@ func (o *Options) rebuildCatalogs(ctx context.Context, dstDir string, filesInArc
 }
 
 func (o *Options) buildCatalogImage(ctx context.Context, ref reference.DockerImageReference, dir string) error {
-	dockerfile := filepath.Join(dir, "index.Dockerfile")
+
+	dockerfile := filepath.Join(o.Dir, "index.Dockerfile")
```

Comment:
For reproducibility purposes, instead of removing the dockerfile how about splitting the `dir` parameter into `dockerfileDir` and `dcDir`?